dev-lang/ghc/ghc-9.0.2-r4.ebuild

```diff
@@ -187,6 +196,11 @@ REQUIRED_USE="
 # haskell libraries built with cabal in configure mode, #515354
 QA_CONFIGURE_OPTIONS+=" --with-compiler --with-gcc"
 
+python_check_deps() {
+	python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" &&
```

Comment:
I suppose we only need these if USE="doc -test" so the condition needs to be tweaked for that?

Reply:
> I suppose we only need these if USE="doc -test" so the condition needs to be tweaked for that?

I'm not 100% sure what you mean. `USE=test` and/or `USE=doc` require python (I assume just `${PYTHON_DEPS}`), but `USE=doc` specifically requires the sphinx stuff regardless if `USE=test` is on.

---

There is this in the ebuild:

https://github.com/gentoo/gentoo/blob/50a0296cec843fe9d6fb095f16360cbfac2e4678/dev-lang/ghc/ghc-9.0.2-r4.ebuild#L182-L187

Later in the ebuild we have this under `pkg_setup`:

https://github.com/gentoo/gentoo/blob/50a0296cec843fe9d6fb095f16360cbfac2e4678/dev-lang/ghc/ghc-9.0.2-r4.ebuild#L462-L464

I wasn't sure if `python_check_deps` would be run unless `python-any-r1_pkg_setup` was run.

---

Would this be better?

```ebuild
python_check_deps() {
	if use doc; then
		python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" &&
		python_has_version "dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]"
	fi
}
```

Reply:
> I suppose we only need these if USE="doc -test" so the condition needs to be tweaked for that?

I'm not 100% sure what you mean. `USE=test` and/or `USE=doc` require python (I assume just `${PYTHON_DEPS}`), but `USE=doc` specifically requires the sphinx stuff regardless if `USE=test` is on.

---

There is this in the ebuild:

https://github.com/gentoo/gentoo/blob/50a0296cec843fe9d6fb095f16360cbfac2e4678/dev-lang/ghc/ghc-9.0.2-r4.ebuild#L182-L187

Later in the ebuild we have this under `pkg_setup`:

https://github.com/gentoo/gentoo/blob/50a0296cec843fe9d6fb095f16360cbfac2e4678/dev-lang/ghc/ghc-9.0.2-r4.ebuild#L462-L464

I wasn't sure if `python_check_deps` would be run unless `python-any-r1_pkg_setup` was run.

---

Would this be better?

```ebuild
python_check_deps() {
	if use doc; then
		python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" &&
		python_has_version "dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]"
	fi
}
```

Reply:
Your suggested `python_check_deps` sounds right.